crates/provider/src/provider/eth_call.rs

```diff
@@ -37,40 +41,69 @@ impl<N: Network> serde::Serialize for EthCallParams<'_, '_, N> {
 }
 
 /// The [`EthCallFut`] future is the future type for an `eth_call` RPC request.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 #[doc(hidden)] // Not public API.
 #[allow(unnameable_types)]
 #[pin_project::pin_project]
-pub struct EthCallFut<'req, 'state, T, N, Resp, Output, Map>(
-    EthCallFutInner<'req, 'state, T, N, Resp, Output, Map>,
-)
+pub struct EthCallFut<'req, T, N, Resp, Output, Map>
 where
     T: Transport + Clone,
     N: Network,
     Resp: RpcReturn,
-    Map: Fn(Resp) -> Output;
+    Output: 'static,
+    Map: Fn(Resp) -> Output,
+{
+    inner: EthCallFutInner<'req, T, N, Resp, Output, Map>,
+    fut: OnceLock<ProviderCall<T, serde_json::Value, Resp>>,
```

Comment:
wondering if moving `fut` into `EthCallFutInner` would make more sense?

Reply:
We can do that, but it would make `EthCallFutInner` non-cloneable. `ProviderCall` cannot be cloned due to the `BoxedFuture` variant.

Reply:
Don't see a scenario where cloning this would be necessary, so +1. Can apply this to `States` and remove `WithBlockFut` as well.

Reply:
yep, it's a private type anyway